scripts/_helpers.py

```diff
@@ -821,3 +829,534 @@ def get_last_commit_message(path):
+
+    candidates = pd.concat(
+        [n.lines[ln_attrs], n.links.loc[n.links.carrier == "DC", lk_attrs]]
+    ).fillna(0)
+
+    positive_order = candidates.bus0 < candidates.bus1
+    candidates_p = candidates[positive_order]
+    swap_buses = {"bus0": "bus1", "bus1": "bus0"}
+    candidates_n = candidates[~positive_order].rename(columns=swap_buses)
+    candidates = pd.concat([candidates_p, candidates_n])
+
+    def make_index(c):
+        return prefix + c.bus0 + connector + c.bus1
+
+    topo = candidates.groupby(["bus0", "bus1"], as_index=False).mean()
+    topo.index = topo.apply(make_index, axis=1)
+
+    if not bidirectional:
+        topo_reverse = topo.copy()
+        topo_reverse.rename(columns=swap_buses, inplace=True)
+        topo_reverse.index = topo_reverse.apply(make_index, axis=1)
+        topo = pd.concat([topo, topo_reverse])
+
+    return topo
+
+
+def cycling_shift(df, steps=1):
+    """
+    Cyclic shift on index of pd.Series|pd.DataFrame by number of steps.
+    """
+    df = df.copy()
+    new_index = np.roll(df.index, steps)
+    df.values[:] = df.reindex(index=new_index).values
+    return df
+
+
+def download_gadm(country_code, update=False, out_logging=False):
+    """
+    Download gpkg file from GADM for a given country code.
+
+    Parameters
+    ----------
+    country_code : str
+        Two letter country codes of the downloaded files
+    update : bool
+        Update = true, forces re-download of files
+
+    Returns
+    -------
+    gpkg file per country
+    """
+
+    gadm_filename = f"gadm36_{two_2_three_digits_country(country_code)}"
+    gadm_url = f"https://biogeo.ucdavis.edu/data/gadm3.6/gpkg/{gadm_filename}_gpkg.zip"
+    _logger = logging.getLogger(__name__)
+    gadm_input_file_zip = get_path(
+        get_current_directory_path(),
+        "data",
+        "raw",
+        "gadm",
+        gadm_filename,
+        gadm_filename + ".zip",
+    )  # Input filepath zip
+
+    gadm_input_file_gpkg = get_path(
+        get_current_directory_path(),
+        "data",
+        "raw",
+        "gadm",
+        gadm_filename,
+        gadm_filename + ".gpkg",
+    )  # Input filepath gpkg
+
+    if not path_exists(gadm_input_file_gpkg) or update is True:
+        if out_logging:
+            _logger.warning(
+                f"Stage 4/4: {gadm_filename} of country {two_digits_2_name_country(country_code)} does not exist, downloading to {gadm_input_file_zip}"
+            )
+        #  create data/osm directory
+        build_directory(gadm_input_file_zip)
+
+        with requests.get(gadm_url, stream=True) as r:
+            with open(gadm_input_file_zip, "wb") as f:
+                shutil.copyfileobj(r.raw, f)
+
+        with zipfile.ZipFile(gadm_input_file_zip, "r") as zip_ref:
+            zip_ref.extractall(get_dirname_path(gadm_input_file_zip))
+
+    return gadm_input_file_gpkg, gadm_filename
+
+
+def get_gadm_layer(country_list, layer_id, update=False, outlogging=False):
```

Comment:
Same as before? (It's in build shapes?)

Reply:
Hey @pz-max , ja same comment as before. I would like to have a dedicated pull request for address this "duplication". Most likely I need to properly unit test both version in order to merge them accordingly.